### PR TITLE
feat: implement xlinectl `user` commands

### PR DESCRIPTION
**Pull request by bsbds**
Based on: #425
Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: xline-client/src/types/auth.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ impl From<AuthenticateRequest> for xlineapi::AuthenticateRequest {
 }
 
 /// Request for `Authenticate`
-#[derive(Debug)]
+#[derive(Debug, PartialEq)]
```
Comment (Phoenix500526) [inline xline-client/src/types/auth.rs:39]: What is this `PartialEq` for?

**Review by Phoenix500526 — COMMENTED** [review]
LGTM

File: xline-client/src/types/auth.rs (hunk 1 above)
Comment (bsbds) [reply xline-client/src/types/auth.rs:39]: explained in https://github.com/xline-kv/Xline/pull/424#discussion_r1326883906

**Review by Phoenix500526 — APPROVED** [review]

**Review by Phoenix500526 — APPROVED** [review]

**Review by themanforfree — APPROVED** [review]
LGTM

**Pull request merged**